source/scripts/InjectedScript/stargazerChainProvider.ts

```diff
@@ -0,0 +1,35 @@
+import { RequestArguments, EIPChainProvider } from '../common/eipChainProvider';
+import { StargazerChain } from './utils';
+
+
+
+/**
+ * EIP 1193 JS Provider
+ * https://eips.ethereum.org/EIPS/eip-1193
+ */
+class StargazerChainProvider extends EIPChainProvider {
+  #chain: StargazerChain;
+
+  constructor(chain: StargazerChain) {
+    super();
+    this.#chain = chain;
+  }
+
+  get version() {
+    return STARGAZER_WALLET_VERSION;
+  }
+
+  request(args: RequestArguments): Promise<unknown> {
+    throw new Error('Method not implemented.');
+  }
```

Comment:
Is this on purpose? Isn't this class called directly (not extended)?

Reply:
Still WIP